.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,74 @@
+name: Nightly integration tests
+
+# Run on request and every day at 3 AM UTC
+on:
+  workflow_dispatch:
+    inputs:
+      syntax_check:
+        type: boolean
+        required: true
+        description: 'Run syntax checker.'
+      target:
+        description: 'Target'
+        required: true
+        default: 'none'
+        type: choice
+        options:
+          - none
+          - quantinuum
+      target_machine:
+        type: string
+        required: false
+        description: 'Target machine (e.g., H1-1E).'
+  schedule:
+    - cron: 0 3 * * *
+
+jobs:
+  integration_test:
+    name: Intergration test
+    runs-on: ubuntu-latest
+    environment: backend-validation
+    container:
+      image: ghcr.io/nvidia/cuda-quantum:latest-hpc
+      options: --user root
```

Comment:
I don't think running as root is necessary anymore.

Reply:
We still need it if we wan to use the `actions/checkout@v3`. As the non-hpc image, it seems this one won't start the container as root, and thus the checkout action won't have permission to create its files.